libpius/path.py

```diff
@@ -99,11 +99,21 @@ def set_tmpdir(dir):
     if sys.platform == "win32":
         TMP = os.environ.get('TEMP')
         return os.path.join(TMP, dir)
-    elif sys.platfrom == "darwin":
-        TMP = '/private/tmp/'
+    elif sys.platform == "darwin":
+        if os.environ.get('USER') == "root":
+            TMP = '/private/tmp/'
+        else:
+            TMP = os.environ.get('TMPDIR')
         return os.path.join(TMP, dir)
     else:
-        TMP = '/tmp/'
+        if os.environ.get('USER') == "root":
+            return os.path.join('/tmp/', dir)
+        elif os.environ.get('XDG_RUNTIME_DIR') != "":
+            return os.path.join(os.environ.get('XDG_RUNTIME_DIR'), dir)
+        elif os.environ.get('TMPDIR') != "":
+            return os.path.join(os.environ.get('TMPDIR'), dir)
+        else:
+            TMP = '/tmp/'
         return os.path.join(TMP, dir)
```

Comment:
Better check for when pius is being installed after being built. If being installed then use hard coded path as there is no TMPDIR or XDG_RUNTIME_DIR for root in linux. After when user runs the pius program it will check and get the path of XDG_RUNTIME_DIR if it exists, else it will check TMPDIR and use hard coded as fallback. 

(NOTE: XDG_RUNTIME_DIR is a secure tmp dir for single user only and is not accessible by group or others, not even root has access. TMPDIR has a possibility to direct towards a secure tmp location, or it can point to /tmp/)